hashring/hashring_test.go

```diff
@@ -310,7 +310,8 @@ func TestLookupNLoopAround(t *testing.T) {
 	ring.AddMembers(members...)
 
 	unique := make(map[valuetype]struct{})
-	ring.tree.LookupNUniqueAt(1, replicaPoint{hash: 0}, unique)
+	orderedUnique := make([]valuetype, 0, 1)
+	ring.tree.LookupNUniqueAt(1, replicaPoint{hash: 0}, unique, &orderedUnique)
```

Comment:
I just added tests. See what you all think. I did have to remove the optimization that copied the servers because it copied them out of order because it copies from a map. I also considered a straight traversal copy, but that would make the first server in the tree hot.